### PR TITLE
[Release, ROCm] release ROCm docker build for AMD MI GPUs

### PR DESCRIPTION
**Pull request by HaiShaw**
## Motivation

Add ROCm docker build to release process, host to `lmsysorg/sglang`

## Modifications

`release-docker.yml`

## Checklist

- [+] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [+] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [+] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Review by zhyncs — COMMENTED** [review]
@HaiShaw have you tested this workflow on your own repo or branch?

**Comment by HaiShaw** [comment]
> @HaiShaw have you tested this workflow on your own repo or branch?

@zhyncs , I didn't test release workflow locally as I don't tag or release locally, and have no token to push `lmsysorg/sglang`.
But the Dockerfile.rocm is well tested.

**Comment by zhyncs** [comment]
@HaiShaw I previously tested it by creating a temporary project on Docker Hub.

**Comment by HaiShaw** [comment]
> @HaiShaw I previously tested it by creating a temporary project on Docker Hub.

@zhyncs sounds good, how do I trigger the workflow?

**Comment by zhyncs** [comment]
https://github.com/sgl-project/sglang/blob/463d56bf4439b078748c47421dffaf73d8eaede4/.github/workflows/release-docker.yml#L8

You can copy release-docker.yml to your repo and then trigger it, but you need to modify the corresponding paths.

**Comment by HaiShaw** [comment]
I did folk sglang to a separate repo, did push the change here with updated path to test dockerhub repo, then to bump up `python/sglang/version.py`.
Did it sound right, I don't have anything special setup on test repo side.

**Comment by zhyncs** [comment]
Nope You should create your own docker repo and set token

**Comment by HaiShaw** [comment]
I did use my own docker hub repo, where to set token?
Any pointer to setup workflow on another github repo, like token?

**Comment by zhyncs** [comment]
GitHub secrets

**Comment by merrymercy** [comment]
Assign this to @zhyncs and please help @HaiShaw to finish the test.

**Comment by zhyncs** [comment]
Yesterday, I synced with @HaiShaw  and found that there is currently an issue with the base image size being too large, which leads to docker build failure.

**Comment by HaiShaw** [comment]
@zhyncs I am not sure whether the base image too large or one layer is too large, as base image to the `Dockerfile` is also `> 7GB`.

**Comment by HaiShaw** [comment]
@zhyncs I may convert this to draft before we know more.

**Comment by HaiShaw** [comment]
@zhyncs Is it possible we can use different runner vs. github default?
Base rocm release docker is somehow already big (~60GB)

**Comment by zhyncs** [comment]
https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/choosing-the-runner-for-a-job

**Comment by zhyncs** [comment]
As discussed offline, for NVIDIA GPUs, we can use the default. For AMD GPUs, we may use self-hosted. Could you update the PR to separate AMD and NVIDIA? Thanks!

**Comment by merrymercy** [comment]
@HaiShaw Let us split this as two jobs in `.github/workflows/release-docker.yml` and let the amd one run on `1-gpu-runner-amd`

File: .github/workflows/release-docker.yml
Comment (merrymercy) [inline .github/workflows/release-docker.yml:1]: create a new file `release-docker-amd.yml`

**Pull request closed without merging**